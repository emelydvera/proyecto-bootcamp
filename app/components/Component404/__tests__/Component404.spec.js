const React = require("react");
const Component404 = require("../Component404");
const { render } = require("@testing-library/react");

describe("Component Error 404", () => {
  const i18n = { gettext: (text) => text };

  it("matches snapshot", () => {
    const { asFragment } = render(<Component404 i18n={i18n} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders an error image", () => {
    const { getByAltText } = render(<Component404 i18n={i18n} />);
    expect(getByAltText("error image")).toBeInTheDocument();
  });

  it("renders error message", () => {
    const { getByText } = render(<Component404 i18n={i18n} />);
    expect(getByText("Parece que esta página no existe")).toBeInTheDocument();
  });

  it("render a redirect button", () => {
    const { getByText } = render(<Component404 i18n={i18n} />);
    expect(getByText("Ir a la página principal")).toBeInTheDocument();
  });

  it("ancore contains link and href attribute", () => {
    const { getByRole } = render(<Component404 i18n={i18n} />);
    expect(getByRole("link")).toHaveAttribute("href", "/");
  });
});
