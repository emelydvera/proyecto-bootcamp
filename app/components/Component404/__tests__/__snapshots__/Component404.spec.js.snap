// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Error 404 matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="error"
  >
    <img
      alt="error image"
      class="lazy-loadable"
      data-src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
    />
    <h4
      aria-label="Parece que esta página no existe"
      class="error__title"
    >
      Parece que esta página no existe
    </h4>
    <a
      class="error__link"
      href="/"
    >
      <button
        aria-label="volver a home"
      >
        Ir a la página principal
      </button>
    </a>
  </div>
</DocumentFragment>
`;
